bokehjs/src/coffee/renderer/glyph/line.coffee

```diff
@@ -33,6 +33,131 @@ define [
       if drawing
         ctx.stroke()
 
+    _hit_point: (geometry) ->
+      ### Check if the point geometry hits this line glyph and return an object
+      that describes the hit result:
+
+        Args:
+          * geometry (object): object with the following keys
+            * vx (float): view x coordinate of the point
+            * vy (float): view y coordinate of the point
+            * type (str): type of geometry (in this case it's a point)
+
+        Output:
+          Object with the following keys:
+            * 0d (bool): whether the point hits the glyph or not
+            * 1d (array(int)): array with the indices hit by the point
+
+      ###
+      [vx, vy] = [geometry.vx, geometry.vy]
+      x = @renderer.xmapper.map_from_target(vx)
+      y = @renderer.ymapper.map_from_target(vy)
+      [x0, x1] = [x-1, x+1]
+      [y0, y1] = [y-1, y+1]
+
+      result = {
```

Comment:
Actually here is something that should be added to the `0d` object: the index of all the points "before" the intersection. that is, if the span hit between puts 7 and 8 then
```
    0d: { flag: true, indices: [7] }
```
That will help with the interpolation later, because for line and patch it really needs to key of the `0d` data, in case the span intersects a line away from any of its points.